### PR TITLE
Docs gp stat progress dtx recovery

### PR DESCRIPTION
**Pull request by mperezfuster**
This PR adds the view `gp_stat_progress_dtx_recovery` to the documentation reference pages for system views. 
Based on https://github.com/greenplum-db/gpdb/pull/15945
Let me know if the column types and descriptions are correct.

### CODE REVIEW

File: gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md [hunk 1]
```diff
@@ -836,6 +838,20 @@ The `gp_stat_progress_create_index_summary` view aggregates across the Greenplum
...
+
+|Column|Type|Description|
+|------|----|-----------|
+|`phase`|text | Status of the recovery. The possible values are: 'initializing', 'recovering commited distributed transactions', 'gathering in-doubt transactions', 'aborting in-doubt transactions', 'gathering in-doubt orphaned transactions', and 'managing in-doubt orphaned transactions'.|
+|`recover_commited_dtx_total`|integer| Total number of commited transactions found to recover.|
+|`recover_commited_dtx_completed`|integer| Total number of in-doubt transaction found, used in startup and non-startup phases.|
+|`in_doubt_tx_in_progress`|integer| Number of in-progress, in-doubt transactions. Its value is always 0 for startup.|
+|`in_doubt_tx_aborted`|integer| Number of aborted in-doubt transactions.|
+
```
Comment (Chibin) [inline gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md:854]: It looks like we're missing one of the columns.
There are total of 6 columns:
 phase,
 recover_commited_dtx_total,
 recover_commited_dtx_completed,
 in_doubt_tx_total,
 in_doubt_tx_in_progress,
 in_doubt_tx_aborted

File: gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md [hunk 2]
```diff
@@ -836,6 +838,20 @@ The `gp_stat_progress_create_index_summary` view aggregates across the Greenplum
...
+This view may be useful if a coordinator restart remains in a recovery state for a long time. Greenplum will not accept connections until all "in-doubt" transactions are resolved. In-doubt transactions are transactions that have been prepared but not committed yet. If there were many transactions running before the coordinator restarted, the recovery of the database may take longer than expected, and you may use this view to monitor the current phase of the recovery. Note that in this scenario, as Greenplum is still starting up, you will need to access the database in utility mode to check this view.
+
+|Column|Type|Description|
+|------|----|-----------|
+|`phase`|text | Status of the recovery. The possible values are: 'initializing', 'recovering commited distributed transactions', 'gathering in-doubt transactions', 'aborting in-doubt transactions', 'gathering in-doubt orphaned transactions', and 'managing in-doubt orphaned transactions'.|
+|`recover_commited_dtx_total`|integer| Total number of commited transactions found to recover.|
+|`recover_commited_dtx_completed`|integer| Total number of in-doubt transaction found, used in startup and non-startup phases.|
```
Comment (Chibin) [inline gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md:851]:
```suggestion
|`in_doubt_tx_total`|integer| Total number of in-doubt transaction found, used in startup and non-startup phases.|
```

File: gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md (hunk 1 above)
Comment (mperezfuster) [reply gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md:854]: Thanks @Chibin it looks like I mixed up two columns. For the column recover_commited_dtx_completed, -- recover completed, this is always 0 after startup.
Is this a boolean? Does it indicate if all committed transactions have been recovered?

Also recover_commited_dtx_total and recover_commited_dtx_total have a typo, it should be committed. Not sure if that can be addressed easily.

Comment (Chibin) [reply gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md:854]: > Thanks @Chibin it looks like I mixed up two columns. For the column recover_commited_dtx_completed, -- recover completed, this is always 0 after startup. Is this a boolean? Does it indicate if all committed transactions have been recovered?

It is not a boolean. It only increases incrementally as each transaction recovery finish.

> 
> Also recover_commited_dtx_total and recover_commited_dtx_total have a typo, it should be committed. Not sure if that can be addressed easily.
Good catch!
It cannot be easily changed at this point... :(

**Review by Chibin — APPROVED** [review]
LGTM!

**Pull request merged**